File: Source/DotNET/Backend/Features/IFeaturesParser.cs
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Dolittle.Vanir.Backend.Features
{
    /// <summary>
    /// Defines a system that can parse features from JSON.
    /// </summary>
    public interface IFeaturesParser
    {
        /// <summary>
        /// Parse a JSON representation.
        /// </summary>
        /// <param name="json">JSON as string to parse.</param>
        /// <returns><see cref="Features"/> instance.</returns>
        Features Parse(string json);
    }
}

File: Source/typescript/features/IFeaturesParser.ts
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { Features } from './Features';

/**
 * Defines a system that can parse features from JSON.
 */
export abstract class IFeaturesParser {

    /**
     * Parse a JSON representation.
     * @param {string} json JSON as string to parse.
     * @returns {Features} instance.
     */
    abstract parse(json: string): Features;
}
